### PR TITLE
Add NCM to networking configuration for OSX Catalina (10.15) compatibility

### LINKED ISSUE #90: Mention in README that Catalina can't initialize the network drivers out of the box
The only way we've found to get the Xebow's networking to work on macOS Catalina is to use https://github.com/jwise/HoRNDIS

And specifically, HoRNDIS also doesn't work out of the box on Cataline, so we had to use this package someone provided in the comments: https://github.com/jwise/HoRNDIS/issues/102#issuecomment-541237232

This should all be mentioned in the README.

### LINKED ISSUE #91: Replace userspace generated USB devices through ConfigFS with something more robust and portable
Nerves out of the box uses a built-in kernel module `CONFIG_USB_ETH=y`. This is widely compatible across operating systems, but lacks the HID device we need for a keyboard.

The conventional wisdom to create composite USB devices is to use Linux's ConfigFS, which is what xebow does. (It creates a composite ECM Ethernet + RNDIS Ethernet + HID device, using ConfigFS: https://github.com/ElixirSeattle/xebow/blob/master/lib/xebow/hid_gadget.ex).  The problem seems to be that we can't create an Ethernet device through ConfigFS that mimics the portability and robustness of the built-in kernel module... the ECM + RNDIS devices are a hack (one works on windows, one works on linux, and neither work on macOS out of the box (#90).

We need to either try harder to mimic the same behavior as the built-in module through configFS, or scrap configFS altogether, and maybe write our own composite HID + Ethernet kernel module based on the one Nerves uses out of the box.

I consider this problem "very hard".

### PR DESCRIPTION
**Pull request by vanvoljg**
This addresses the issue of getting the Xebow to connect to a Mac running Catalina, OSX 10.15. The fix adds an NCM function to the ConfigFS-based USB gadget that we set up to make everything work over one USB port.

This is dependent on [PR #3 of `nerves_system_keybow`](https://github.com/ElixirSeattle/nerves_system_keybow/pull/3) getting merged in, along with generation of a new release tag and Nerves artifact.

I went and left ECM in to increase compatibility, in case some hosts are unable to use the other modes.

@doughsay could you give it a whirl on your Mac again to make sure that my adding ECM back in didn't mess anything up?

It can always get taken out again, if it causes issues.

- [x] Fix `nerves_system_keybow` dependency version

Fixes #90
Fixes #91

### CODE REVIEW

File: mix.exs [hunk 1]
```diff
@@ -105,13 +105,15 @@ defmodule Xebow.MixProject do
       {:circuits_gpio, "~> 0.4", targets: @all_targets},
       {:circuits_spi, "~> 0.1", targets: @all_targets},
       {:usb_gadget, github: "nerves-project/usb_gadget", ref: "master", targets: @all_targets},
+      {:observer_cli, "~> 1.5"},
```
[the remark this answers is not in the document]
Comment (vanvoljg) [reply mix.exs:108]: Kind of hoped nobody would notice that... It's the only means of viewing the device's opinion of transfer speed.

I'll take it out when I update the `nerves_system_keybow` dependency.

**Comment by doughsay** [comment]
> Exposing 3 ethernet endpoints to the host simultaneously feels excessive and brittle to me

I agree, it already annoys me that I have two interfaces that show up on linux, and one doesn't work.  It always tries to re-connect, causing notification spam. And what happens if two work?  If both connect successfully what happens?

> What if rather than bonding the adapters in the device, check which adapters can communicate with the host and disable all of the ones that are not the preferred adapter

This would be amazing and nullify the concern above. But I'm not sure how to pull it off...

File: mix.exs (hunk 1 above)
[the remark this answers is not in the document]
Comment (amclain) [reply mix.exs]: I'm not necessarily against having `observer_cli` in the project, but if we do add it, it should probably be scoped to a different PR.

**Comment by doughsay** [comment]
@vanvoljg there are still some changes left in here that reverence usb2, but looks like you opted to remove ecm in the end?

**Comment by vanvoljg** [comment]
> run mix deps.get and check in the mix.lock changes

I knew there was something I'd forgotten. Cheers.

> @vanvoljg there are still some changes left in here that reverence usb2, but looks like you opted to remove ecm in the end?

Comment documentation? Fixed! Thanks for the reminder. Any other references to usb2 are for the HID device.

**Pull request merged**